as/plugins/org.jboss.ide.eclipse.as.core/plugin.xml

```diff
@@ -3020,5 +3020,77 @@
+            description="%jboss.version.wildfly24.description"
+            hasConfiguration="false"
+            initialState="stopped"
+            launchConfigId="org.jboss.ide.eclipse.as.core.server.startupConfiguration"
+            launchModes="run,debug,profile"
+            name="%jboss.version.wildfly24.name"
+            runtime="false"
+            runtimeTypeId="org.jboss.ide.eclipse.as.runtime.wildfly.240"
+            startBeforePublish="false"
+            startTimeout="450000"
+            stopTimeout="450000"
+            supportsRemoteHosts="true"/>
+   </extension>
+
+   <extension point="org.eclipse.wst.server.core.runtimeTypes">
+      <runtimeType
+             vendor="%providerName"
+             class="org.jboss.ide.eclipse.as.core.server.internal.v7.LocalWildfly80ServerRuntime"
+             description="%jboss.version.wildfly24.description"
+             name="%jboss.version.wildfly24.runtime.name"
+             id="org.jboss.ide.eclipse.as.runtime.wildfly.240"
+             version="24.0">
+          <moduleType types="jst.appclient" versions="5.0, 6.0,7.0,8.0"/>
+          <moduleType types="wst.web" versions="1.0,1.2,1.3,1.4"/>
+          <moduleType types="jst.web" versions="2.2, 2.3, 2.4, 2.5, 3.0, 3.1, 4.0"/>
+          <moduleType types="jst.ejb" versions="1.0, 1.1, 2.0, 2.1, 3.0, 3.1, 3.2"/>
+          <moduleType types="jst.ear" versions="1.2, 1.3, 1.4, 5.0, 6.0, 7.0, 8.0"/>
+          <moduleType types="jst.connector" versions="1.0, 1.5, 1.6, 1.7"/>
+          <moduleType types="jst.utility" versions="1.0"/>
+          <moduleType types="jboss.package" versions="1.0"/>
+          <moduleType types="jboss.singlefile" versions="1.0"/>
+          <moduleType types="jst.jboss.sar" versions="1.0"/>
+          <moduleType types="jbt.bpel.module" versions="1.1, 2.0"/>
+          <moduleType types="bpel.module" versions="1.1, 2.0"/>
+          <moduleType types="jboss.osgi" versions="1.0"/>
+      </runtimeType>
+   </extension>
+   <extension point="org.eclipse.jst.server.core.runtimeFacetMappings">
+      <runtimeFacetMapping
+            runtimeTypeId="org.jboss.ide.eclipse.as.runtime.wildfly.240"
+            runtime-component="org.jboss.ide.eclipse.as.runtime.wildfly.component"
+            version="24.0"/>
+   </extension>
+	<extension
+         point="org.eclipse.wst.common.project.facet.core.runtimes">
+      <runtime-component-type id="org.jboss.ide.eclipse.as.runtime.wildfly.component"/>
+      <runtime-component-version
+            type="org.jboss.ide.eclipse.as.runtime.wildfly.component"
+            version="24.0"/>
+     <supported>
+        <runtime-component
+         	id="org.jboss.ide.eclipse.as.runtime.wildfly.component"
+         	version="24.0"/>
+       <facet id="wst.web"  version="1.0,1.2,1.3,1.4"/>
+       <facet id="jst.web"  version="2.2,2.3,2.4,2.5,3.0,3.1,4.0"/>
+       <facet id="jst.java" version="5.0,6.0,1.7,1.8,9,10,11,12"/>
```

Comment:
Shouldn't we go til Java 16 as it is supported ?

Reply:
Added up to java16. Thanks.